js/ui/components/clubhouse.js

```diff
@@ -195,6 +199,62 @@ var ClubhouseNotificationBanner = new Lang.Class({
         }
     },
 
+    _rearrange_elements: function() {
+        let contentBox = this._bodyStack.get_parent();
+        contentBox.add_style_class_name('clubhouse-content-box');
+        let hbox = contentBox.get_parent();
+        let vbox = hbox.get_parent();
+
+        let wrapBin = new St.Bin({ x_fill: true,
+                                   y_fill: true,
+                                   x_align: St.Align.END,
+                                   y_align: St.Align.END,
+                                   x_expand: true,
+                                   y_expand: true });
+        hbox.add_child(wrapBin);
+
+        // A Clutter.BinLayout is used to rearrange the notification
+        // elements in layers:
+        let wrapWidget = new St.Widget({ layout_manager: new Clutter.BinLayout(),
+                                         x_align: Clutter.ActorAlign.FILL,
+                                         y_align: Clutter.ActorAlign.END });
+        wrapBin.set_child(wrapWidget);
+
+        this._iconBin.set_fill(false, false);
+        this._iconBin.set_x_expand(true);
+        this._iconBin.set_y_expand(true);
+        this._iconBin.set_x_align(Clutter.ActorAlign.END);
+        this._iconBin.set_y_align(Clutter.ActorAlign.END);
+
+        contentBox.set_x_expand(true);
+        contentBox.set_y_expand(false);
+        contentBox.set_x_align(Clutter.ActorAlign.FILL);
+        contentBox.set_y_align(Clutter.ActorAlign.END);
+
+        this._actionBin.set_x_expand(true);
+        this._actionBin.set_y_expand(true);
+        this._actionBin.set_x_align(Clutter.ActorAlign.START);
+        this._actionBin.set_y_align(Clutter.ActorAlign.END);
+
+        this._expandedLabel.actor.set_y_expand(true);
+        this._expandedLabel.actor.set_y_align(Clutter.ActorAlign.CENTER);
+        this._expandedLabel.actor.clutter_text.line_wrap_mode = Pango.WrapMode.WORD_CHAR;
+
+        hbox.remove_child(contentBox);
+        wrapWidget.add_child(contentBox);
+
+        hbox.remove_child(this._iconBin);
+        wrapWidget.add_child(this._iconBin);
+
+        vbox.remove_child(this._actionBin);
+        wrapWidget.add_child(this._actionBin);
```

Comment:
I'd also like to discuss this with you more directly on Slack.

Reply:
Sounds good. Otherwise let me know here if there is any other way in a message here.